components/service/experiments/src/main/java/mozilla/components/service/experiments/ExperimentsUpdater.kt

```diff
@@ -25,23 +29,55 @@ internal class ExperimentsUpdater(
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal lateinit var source: KintoExperimentSource
 
-    @Volatile private var updateJob: Job? = null
-
     internal fun initialize(configuration: Configuration) {
         source = getExperimentSource(configuration)
 
-        updateJob = GlobalScope.launch(IO) {
-            // Update the experiments list from the server async, without blocking
-            // the app launch.
-            updateExperiments()
-        }
+        // Schedule the periodic experiment updates
+        scheduleUpdates()
```

Comment:
Per IRC conversation, we currently don't guarantee immediate updates on app launch.

We can change this to use .REPLACE, then we get:
- An immediate update on app launch.
- A predictable schedule (it resets to update every 6h from app launch again)